hack/verify-generated.sh

```diff
@@ -18,21 +18,9 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-HELM_DOCS_BIN=${1:-}
-
-if [[ -z $HELM_DOCS_BIN ]]; then
-	echo "usage: $0 <path-to-helm-docs>"
-	exit 1
-fi
-
-TEMP_FILE=$(mktemp)
-trap '{ rm -f -- "$TEMP_FILE"; }' EXIT
-
-$HELM_DOCS_BIN ./deploy/charts/trust-manager -d -l error > ${TEMP_FILE}
-
-if ! cmp -s "./deploy/charts/trust-manager/README.md" "${TEMP_FILE}"; then
-  echo "Helm chart README.md is out of date."
-  echo "Please run 'make update-helm-docs'."
+git --no-pager diff
+git status --short --untracked-files=no
+if [ ! -z "$(git status --short --untracked-files=no)" ]; then
+  echo -e "\e[31mGenerated files do not match those committed. Please run 'make generate-all' locally and commit the changes. \e[0m"
```

Comment:
comment: As I'm sure you know, this is a stronger check than just checking if the generated files are out of date - but I think it's a check we should make anyway. Ideally files shouldn't be getting modified in CI so in CI it's good to verify they're not.

I might be wrong, but I think this would also fail if I the code generation was up-to-date, I'd edited something unrelated to code generation and I ran `make verify` locally, as part of development? That seems unfortunate - I use targets like `make verify` quite often when I'm developing a new feature.

This new way of validating would then mean I'd have to commit my changes before CI could pass. I suppose that's not the end of the world!

Reply:
True, but outdated comment now. 😉